scripts/globals/spells/geo-poison.lua

```diff
@@ -0,0 +1,28 @@
+-----------------------------------------
+-- Spell: Geo-Poison
+-----------------------------------------
+require("scripts/globals/status")
+require("scripts/globals/geo")
+-----------------------------------------
+
+function onMagicCastingCheck(caster, target, spell)
+    if caster:getPet() ~= nil then
+        return tpz.msg.basic.LUOPAN_ALREADY_PLACED
+    elseif not caster:canUseMisc(tpz.zoneMisc.PET) then
+        return tpz.msg.basic.CANT_BE_USED_IN_AREA
+    else
+        return 0
+    end
+end
+
+
+function onSpellCast(caster, target, spell)
+    local geo_skill = caster:getCharSkillLevel(tpz.skill.GEOMANCY)
+    local spellCost = 37
+    local power = (geo_skill / 30) / 10
```

Comment:
Geomancy power is a combination of both Geomancy skill _and_ Handbell skill if a handbell was used during casting of the spell. (Hence why subbing GEO is a bad idea, because only GEO can equip bells.)

I think what we'll want to do with these in the future is have a helper in `tpz.geo` that takes a caster and a skill cap, and returns: potency, geomancy+ total. The helper will handle getting and combining the appropriate skill levels, cap it off if necessary, and return the caster's applicable power and geomancy+ mod. Each individual spell script can then use that information to determine the power of their effects and spawn the bubble/luopan.

On the subject of Poison bubbles, bgwiki says it caps out at 30 dmg/tick if at 600 skill.
`(600 / 30) / 10 = 2`, which is either not right, or I'm missing something here~ 😅 

Also, do we _not_ have spell:getCost()? If not, we'll probably want to add one.

Reply:
this should be all good if you look at the table in geo.lua, all retail values.